isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+#include "Isis.h"
+
+#include "Application.h"
+#include "Cube.h"
+#include "CubeAttribute.h"
+#include "FileName.h"
+#include "Portal.h"
+#include "ProcessByLine.h"
+#include "Pvl.h"
+#include "SpecialPixel.h"
+#include "UserInterface.h"
+
+#include <QString>
+
+using namespace std;
+using namespace Isis;
+
+// Taken from ProcessMosaic
+const int FLOAT_MIN = -16777215;
+const int FLOAT_MAX = 16777216;
+
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand);
+void createMosaicCube(UserInterface ui, QString bands);
+void createTrackCube(UserInterface ui, QString trackingBand);
+void copyPixels(Buffer &in, Buffer &out);
+void copyTrackPixels(Buffer &in, Buffer &out);
+
+void IsisMain() {
+  UserInterface &ui = Application::GetUserInterface();
+  QString copyBands;
+  QString trackBand;
```

Comment:
It feels really weird passing a list of integers around as a string. I'd much rather see these as lists of integers

Reply:
I store them as a QString because CubeAttributeOutput takes in a string of bands to copy over from the input cube. If I stored them in a list, I would have to create another for loop to store them in a QString.

Reply:
My main concern is that it's completely opaque that this may be a list of bands. Using a QStringList would fix that without much difficulty. You can just add the band indices as you get them, you wouldn't need line 60 anymore. Then, when you add them to the filename you can use QStringList::join(",") to put them together.

Reply:
I did not realize, but CubeAttributeInput has a setBands method that takes in a vector of QStrings. I'll go ahead and change it to that. Thanks!